contrib/integrations/artifactory/artifactory.go

```diff
@@ -257,100 +258,136 @@ func PrepareBuildInfo(ctx context.Context, artiClient artifact_manager.ArtifactM
+func prepareBuildInfo(ctx context.Context, client artifact_manager.ArtifactManager, r sdk.WorkflowRunResult, execContext executionContext) (*buildinfo.Module, error) {
+	ctx, endc := telemetry.Span(ctx, "artifactory.PrepareBuildInfo", telemetry.Tag("runResult.Type", r.Type))
+	if r.Type != sdk.WorkflowRunResultTypeArtifactManager {
+		endc()
+		return nil, nil
+	}
+
+	var currentMaturity string
+	if r.DataSync != nil {
+		latestPromotion := r.DataSync.LatestPromotionOrRelease()
+		if latestPromotion != nil {
+			currentMaturity = latestPromotion.ToMaturity
 		}
+	}
+	if currentMaturity == "" {
+		currentMaturity = execContext.defaultLowMaturitySuffix
+	}
 
-		artifacts, err := retrieveModulesArtifacts(ctx, client, files)
+	data, err := r.GetArtifactManager()
+	if err != nil {
+		endc()
+		return nil, err
+	}
+	mod := buildinfo.Module{
+		Id:           fmt.Sprintf("%s:%s", data.RepoType, data.Name),
+		Artifacts:    make([]buildinfo.Artifact, 0),
+		Dependencies: nil,
+	}
+	switch data.RepoType {
+	case "docker":
+		mod.Type = buildinfo.Docker
+		props := make(map[string]string)
+		parsedUrl, err := url.Parse(client.GetURL())
 		if err != nil {
 			endc()
-			return nil, err
+			return nil, sdk.WrapError(err, "unable to parse artifactory url [%s]: %v", client.GetURL())
 		}
-		mod.Artifacts = artifacts
-		modules = append(modules, mod)
+		urlArtifactory := parsedUrl.Host
+		if parsedUrl.Port() != "" {
+			urlArtifactory += ":" + parsedUrl.Port()
+		}
+		props["docker.image.tag"] = fmt.Sprintf("%s.%s/%s", data.RepoName, urlArtifactory, data.Name)
+		mod.Properties = props
+	}
+
+	files, err := retrieveModulesFiles(ctx, client, data.RepoName, data.Path)
+	if err != nil {
 		endc()
+		return nil, err
 	}
 
-	return modules, nil
+	props := utils.NewProperties()
+	props.AddProperty("build.name", fmt.Sprintf("%s/%s/%s", execContext.buildInfo, execContext.projectKey, execContext.workflowName))
+	props.AddProperty("build.number", execContext.version)
+	props.AddProperty("build.timestamp", strconv.FormatInt(time.Now().Unix(), 10))
+
+	if err := SetPropertiesRecursive(ctx, client, data.RepoName, currentMaturity, files, props); err != nil {
+		endc()
+		return nil, err
+	}
+
+	artifacts, err := retrieveModulesArtifacts(ctx, client, files)
+	if err != nil {
+		endc()
+		return nil, err
+	}
+	mod.Artifacts = artifacts
+	endc()
+	return &mod, nil
 }
 
 func retrieveModulesFiles(ctx context.Context, client artifact_manager.ArtifactManager, repoName string, path string) ([]sdk.FileInfo, error) {
 	ctx, end := telemetry.Span(ctx, "workflow.retrieveModulesFiles")
 	defer end()
 	log.Debug(ctx, "retrieve:ModulesFiles repoName:%s path:%s", repoName, path)
+	_, endc := telemetry.Span(ctx, "artifactoryClient.GetFileInfo", telemetry.Tag("path", path), telemetry.Tag("repoName", repoName))
+	fileInfo, err := client.GetFileInfo(repoName, path)
+	endc()
+	if err != nil {
+		return nil, err
+	}
+
+	// If it can be downloaded, it's a file
+	if fileInfo.DownloadURI != "" {
```

Comment:
I think the root path of an artifact is always a folder.

Reply:
no, only for docker